apps/store/src/components/CartInventory/useCampaign.ts

```
@@ -1,20 +1,21 @@
 import { useRedeemCampaignMutation, useUnredeemCampaignMutation } from '@/services/apollo/generated'
-import { useGetMutationError } from '@/utils/useGetMutationError'
+import { useErrorMessage } from '@/utils/useErrorMessage'
 
 type Params = {
   cartId: string
 }
 
 export const useRedeemCampaign = ({ cartId }: Params) => {
-  const getMutationError = useGetMutationError()
-  const [redeemCampaign, result] = useRedeemCampaignMutation()
+  const [redeemCampaign, result] = useRedeemCampaignMutation({
+    onError: () => {}, // Handled via errorMessage
```

why do we need this one? Will it throw an exception otherwise?

---

Exactly, it'll throw if error is not handled.  Here we know that error display is taken care of declaratively and callback is not needed